### PR TITLE
SET uses compute functions

### LINKED ISSUE #269: Make SET relations more compatible with COMPUTE operations
SET operates on each object in a collection and COMPUTE outputs ONE variable. They are not equivalent. If you want to make an operation on one property in each object you need SET. A lot of functions are missing.

I added some in PR #268  but there is still a huge gap between SET and COMPUTE

### PR DESCRIPTION
**Pull request by larkob**
SET now uses the operations from COMPUTE for the `relation` parameter. The property value is mapped to `operand1` and the value to `operand2`. There's no breaking changes, but you can now do stuff like 
```
    {
      "func": "SET",
      "collection": "thisButton",
      "property": "x",
      "relation": "pow",
      "value": 2
    }
```

Other changes:
* added "random" to operations
* added "=" to operations, which simply sets a property to the given value (no computation, for backward compatibility with SET)

Resolves #269 and #270

### CODE REVIEW

**Comment by ArnoldSmith86** [comment]
PR-SERVER-BOT: You can play around with it here: http://212.47.248.129:3279/pr-test (or any other room on that server)

**Comment by RaphaelAlvez** [comment]
Does this also include random to compute?

**Review by RaphaelAlvez — APPROVED** [review]
Very simple and smart code change. I have to admit that the approach I was going to take was way worse than this. I also tested with The demo I created for my PR and it works. Approved

File: client/js/widgets/button.js [hunk 1]
```diff
@@ -255,61 +250,68 @@ export class Button extends Widget {
...
+            v = x[o](y);
             break;
           case 'slice':
-            variables[v] = x[a.operation](y, z);
+            v = x[o](y, z);
             break;
           case 'push':
           case 'unshift':
-            variables[v][a.operation](x);
+            if (Array.isArray(v)) v[o](x);
```
Comment (ArnoldSmith86) [inline client/js/widgets/button.js:294]: Why do you add a check here but nowhere else? Many other operations should have the same problem which should already be caught and added to `problems` anyway.

File: client/js/widgets/button.js [hunk 2]
```diff
@@ -255,61 +250,68 @@ export class Button extends Widget {
...
+          v = 0;
           problems.push(`The operation evaluated to null, Infinity or NaN. Setting the variable to 0.`);
         }
+        return v;
+      }
+
+      if(a.func == 'COMPUTE') {
+        setDefaults(a, { operation: '+', operand1: 1, operand2: 1, operand3: 1, variable: 'COMPUTE' });
+        const toNum = s=>typeof s == 'string' && s.match(/^[-+]?[0-9]+(\.[0-9]+)?$/) ? +s : s;
+        const v = a.variable;
```
Comment (ArnoldSmith86) [inline client/js/widgets/button.js:313]: Not really needed anymore but whatever..

**Review by ArnoldSmith86 — APPROVED** [review]
I did not test it at all but the code looks fine to me.

**Pull request merged**